tests/test_checkpoint.py

```diff
@@ -130,3 +130,16 @@ def test_get_loader_raises_error():
     filepath = "some/wrong/path"
     with pytest.raises(ValueError, match=r"Given path:"):
         EstimatorCheckpoint._get_loader_from_path(filepath)
+
+
+def test_create_writer_when_checkpoint_dir_none():
```

Comment:
It's better to add another test for `checkpoint_dir.startswith(const.GCS_BASE_STR)` case.

Reply:
It was handled in a test above, but I have broken down it to individual test cases.